src/librustc_typeck/astconv.rs

```diff
@@ -327,7 +332,6 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     let mut err = tcx.sess.struct_span_err(span, msg);
                     err.span_note(span_late, note);
                     err.emit();
-                    reported_late_bound_region_err = Some(true);
```

Comment:
`reported_late_bound_region_err.unwrap_or(false)` below would suggest that only this branch should have `reported_late_bound_region_err = true;`, not the lint one below?

Reply:
I can change the variable name. I thought it was reasonable to include emission of a lint as "reported", though, so the name still seemed appropriate.

Reply:
Doesn't that change the behavior later on (potentially preventing real errors from being emitted)?

The important thing, I thought, is whether there was an actual error that would stop compilation (maybe we should use `ErrorReported` to indicate that?).

Reply:
It's possible there are interactions with `impl Trait` that aren't being tested at all. I'll check this.

Reply:
I more or less really want to keep the preexisting behavior, if the tests pass either way.